webapp/views/wpt-results.js

```diff
@@ -1004,6 +1033,46 @@ class WPTResults extends WPTColors(WPTFlags(PathInfo(LoadingState(TestRunsUIBase
     // % in js is not modulo, it's remainder. Ensure it's positive.
     this.path = this.searchResults[(n + next) % n].test;
   }
+
+  computeTestPlural(selectedMetadata) {
+    return this.pluralize('test', selectedMetadata.length);
+  }
+
+  clearSelectedCells(selectedMetadata) {
+    if (selectedMetadata.length === 0 && this.selectedCells.length) {
+      for (const cell of this.selectedCells) {
+        cell.removeAttribute('selected');
+      }
+      const toast = this.shadowRoot.querySelector('#selected-toast');
+      toast.hide();
+      this.selectedCells = [];
+    }
+  }
+
+  handleSelectMetadata(e) {
+    const td = e.target.closest('td');
+    var test = td.getAttribute('data-test');
+    const browser = this.products[td.getAttribute('data-index')].browser_name;
+    if (this.computePathIsASubfolder(test)) {
+      test = test + '/*';
+    }
+
+    if (this.selectedMetadata.find(s => s.test === test && s.product === browser)) {
+      this.selectedMetadata = this.selectedMetadata.filter(s => !(s.test === test && s.product === browser));
+      td.removeAttribute('selected');
```

Comment:
Shall we remove this cell from `this.selectedCells` here? I know it doesn't make a difference (we'll simply try to unset the property again in `clearSelectedCells`), but this can be a footgun in the future if one starts to rely on this property being up to date.